src/m365/spo/commands/page/page-copy.ts

```diff
@@ -0,0 +1,163 @@
+    const telemetryProps: any = super.getTelemetryProperties(args);
+    telemetryProps.sourceName = typeof args.options.sourceName !== 'undefined';
+    telemetryProps.targetUrl = typeof args.options.targetUrl !== 'undefined';
+    telemetryProps.overwrite = !!args.options.overwrite;
+    return telemetryProps;
+  }
+
+  public commandAction(logger: Logger, args: CommandArgs, cb: () => void): void {
+    let sourceFullName: string = args.options.sourceName.toLowerCase();
+    const targetPageInfo: { siteUrl: string, pageName: string } = this.getSiteUrl(args.options.webUrl, args.options.targetUrl.toLowerCase());
+    let { siteUrl: targetSiteUrl, pageName: targetFullName } = targetPageInfo;
+
+    if (sourceFullName.indexOf('.aspx') < 0) {
+      sourceFullName += '.aspx';
+    }
+    if (targetFullName.indexOf('.aspx') < 0) {
+      targetFullName += '.aspx';
+    }
+
+    if (targetSiteUrl.endsWith('/')) {
+      targetSiteUrl = targetSiteUrl.substring(0, targetSiteUrl.length - 1);
+    }
+    if (targetFullName.startsWith('/')) {
+      targetFullName = targetFullName.substring(1);
+    }
+
+    if (this.verbose) {
+      logger.logToStderr(`Creating page copy...`);
+    }
+
+    const requestOptions: any = {
+      url: `${args.options.webUrl}/_api/SP.MoveCopyUtil.CopyFileByPath()`,
+      headers: {
+        'accept': 'application/json;odata=nometadata'
+      },
+      data: {
+        srcPath: { DecodedUrl: `${args.options.webUrl}/sitepages/${sourceFullName}` },
+        destPath: { DecodedUrl: `${targetSiteUrl}/sitepages/${targetFullName}` },
+        options: { ResetAuthorAndCreatedOnCopy: true, ShouldBypassSharedLocks: true },
+        overwrite: !!args.options.overwrite
+      },
+      responseType: 'json'
+    };
+
+    request
+      .post<void>(requestOptions)
+      .catch((err: any): void => {
+        // The API returns a 400 when file already exists
+        if (err && err.response && err.response.status && err.response.status === 400) {
+          return;
+        }
+        // Throw the error if something else happened
+        throw err;
+      })
+      .then((): Promise<ClientSidePageProperties> => {
+        const requestOptions: any = {
+          url: `${targetSiteUrl}/_api/sitepages/pages/GetByUrl('sitepages/${targetFullName}')`,
+          headers: {
+            'accept': 'application/json;odata=nometadata'
+          },
+          responseType: 'json'
+        };
+
+        return request.get<ClientSidePageProperties>(requestOptions)
+      })
+      .then((res: ClientSidePageProperties): void => {
+        logger.log(res);
+
+        if (this.verbose) {
+          logger.logToStderr(chalk.green('DONE'));
+        }
+
+        cb();
+      })
+      .catch((err: any): void => {
+        return this.handleRejectedODataJsonPromise(err, logger, cb)
+      });
+  }
+
+  public options(): CommandOption[] {
+    const options: CommandOption[] = [
+      {
+        option: '--sourceName <sourceName>',
```

Comment:
Since the `sourceName` can be both a name and a site-relative URL, similarly to `targetUrl`, shall we align their naming and use `sourceUrl` instead?

Reply:
`sourceName` is a page on the referenced site, the `targetUrl` property can be on the current site or another site collection. So for the source, I kept it like it is with the other page commands. Used `source` to make it easier to understand instead of using `name` or pageName`.

Reply:
The `source` part makes sense. What's confusing to me is the `Name` part since it's also ok to specify a site-relative URL, so using `sourceUrl` instead of `sourceName` would be clearer

Reply:
@waldekmastykarz there is nothing different than the other page commands which all use `--name` or `--pageName` command here. They can also be used as site-relative URLs.

Reply:
Good point! In that case let's use `sourceName` for now and we can revise it in v4 as it would be a breaking change.

Reply:
Also checked the `spo file` command, there the `--url` property is used. Would indeed be good for the next version to check all of them and align.